stl/inc/cmath

```diff
@@ -1304,6 +1315,40 @@ template <class _Ty>
 _NODISCARD constexpr _Ty _Common_lerp(const _Ty _ArgA, const _Ty _ArgB, const _Ty _ArgT) noexcept {
     // on a line intersecting {(0.0, _ArgA), (1.0, _ArgB)}, return the Y value for X == _ArgT
 
+#ifdef _M_FP_STRICT
+    class _ExceptGuard {
+    public:
+        constexpr _ExceptGuard() {
+            if (!_STD is_constant_evaluated()) {
+                [[maybe_unused]] const int _HoldExcept = feholdexcept(&_Env);
+            }
+        }
+
+        _ExceptGuard(const _ExceptGuard&) = delete;
+        _ExceptGuard& operator=(const _ExceptGuard&) = delete;
+
+        constexpr ~_ExceptGuard() {
+            if (!_STD is_constant_evaluated()) {
+                [[maybe_unused]] const int _UpdateEnv = feupdateenv(&_Env);
+            }
+        }

```

Comment:
`feholdexcept` and `feupdateenv` are quite costly on x86 CPUs. The use case here is one of the scenarios IEEE 754 exceptions were originally designed for, but I think we'd better avoid them for short computations like `lerp` because of performance.

As an example, floating-point `midpoint` checks whether `_Val1 + _Val2` could possibly overflow before actually adding them.
https://github.com/microsoft/STL/blob/18c12ab01896e73e95a69ceba9fbd7250304f895/stl/inc/numeric#L632-L648

I did a simple x64 benchmark on AMD Ryzen 3700X. `a`, `b`, and `t` are random `double`s within [0, 1]:
Algorithm | Time/ns
:---: | :---:
Naïve `a * (1-t) + b * t` | &#x2007;&#x2007;0.493
`std::lerp` before PR (https://github.com/microsoft/STL/commit/18c12ab01896e73e95a69ceba9fbd7250304f895) | &#x2007;&#x2007;7.35&#x2007;
`std::lerp` after PR (https://github.com/microsoft/STL/pull/1918/commits/eb6cc26fc97a057c7540a91abe6c0b3af82087ce) | 135&#x2008;&#x2007;&#x2007;&#x2007;